lambda/src/main/scala/pricemigrationengine/handlers/AmendmentHandler.scala

```diff
@@ -52,6 +52,16 @@ object AmendmentHandler extends CohortHandler {
           val result = ExpiringSubscriptionResult(item.subscriptionName)
           CohortTable.update(CohortItem.fromExpiringSubscriptionResult(result)).as(result)
         }
+        case e: ZuoraUpdateFailure => {
+          // We are only interested in the ZuoraUpdateFailures corresponding to message
+          // "Operation failed due to a lock competition"
+          // We succeed them without cohort item update to be done later.
+          if (e.reason.contains("lock competition")) {
```

Comment:
The reason field is very unlikely to change but if there's anything like a message ID that might be more reliable.

Reply:
The error code is "53500050". Zuora doesn't have an entry for that one, but the documentation says that the way error codes are built, the last two digits are relevant and "50" occurs when a _Locking contention_.

_This request cannot be processed because the objects this request is trying to modify are being modified by another API request, UI operation, or batch job process._

I could switch to search for the entire code inside the string but that feels like as arbitrary as the pattern. Going to keep the pattern for the moment 🙂